Added fix for maxAge string unhandled type error

PaulRBerg opened the pull request:
This solves #432. I'm setting the `maxAge` with an environment variable which is parsed as a string. The PR adds a small fix to automatically convert strings to integers, as the error message is vague and creates confusion.

PaulRBerg commented:
Just implemented the changes now! I thank you for this amazing module.

dougwilson commented:
Awesome, thank you! The indention seems all messed up in the changes, and I'm not sure why the CI didn't catch that. I need to take a look into this as soon as I get to a computer 👍

PaulRBerg commented:
I submitted again now, with proper tabs. Thing is that while I was coding, eslint shouted that it wants spaces. Perhaps just some cross-machine issue.

dougwilson commented:
Yea, indentation is supposed to be spaces.

dougwilson commented on test/cookie.js:
```diff
@@ -39,6 +39,17 @@ describe('new Cookie()', function () {
+      var cookie = new Cookie({ maxAge: 7200000 })
+		  assert.equal(typeof cookie.maxAge, 'number')
+		  assert.equal(cookie.maxAge % 1, 0)
+	  })
```
There is definitely something wrong with either eslint for not catching this or Github is not displaying this right. This closing brace should be aligned with the "it"

dougwilson commented on test/cookie.js:
```diff
@@ -39,6 +39,17 @@ describe('new Cookie()', function () {
+	  })
+
+    it('should reject non-integer maxAge options', function() {
+	    assert.throws(function() { new Cookie({ maxAge: '7200000'}) })
```
And these inner lines should only br 2 spaces indented from the prior level. Is it right on your side and it is just that github is messed up? Because eslint should be catching this issue if it's really there which is very confusing.

dougwilson commented:
Ok, I figured out what the indentation issue is. I updated `master` so `eslint` will now detect the issue 👍

PaulRBerg commented:
I updated `.eslint` and repushed now. However, the issues from above simply do not appear on my machine, even after I added the rule. It might be just GitHub.

dougwilson commented:
Now github looks right. Did you make any changes besides adding the eslint rule?

PaulRBerg commented:
I did, but the code didn't look as above.

dougwilson commented:
> but the code didn't look as above

Hm?

PaulRBerg commented:
I mean it was totally fine, except it had mixed tabs with spaces. Perhaps GitHub handles this case in a special way.

dougwilson commented:
When I was on my machine, it looked just as messed up in my editor as it did in Github, which is how I found out what was happening. So it wasn't just github.

PaulRBerg commented:
Haha then I'll never use Atom again! I'm usually working in IntelliJ, but I wanted to separate this project now.

PaulRBerg commented:
Any updates on this? Checked out the project in IntelliJ and the indentation looks good.

dougwilson commented:
I am rebasing the PR now. Apparently `maxAge` can actually be a `Date` object. I'll update the PR to let that through and the next minor I'll just deprecate that feature to be removed in the future, since it's a weird feature.
